builder/dockerfile/builder.go

```diff
@@ -118,13 +122,20 @@ func NewBuilder(clientCtx context.Context, config *types.ImageBuildOptions, back
 		Stdout:        os.Stdout,
 		Stderr:        os.Stderr,
 		docker:        backend,
-		context:       buildContext,
+		source:        source,
 		runConfig:     new(container.Config),
 		tmpContainers: map[string]struct{}{},
 		buildArgs:     newBuildArgs(config.BuildArgs),
 		escapeToken:   parser.DefaultEscapeToken,
 	}
 	b.imageContexts = &imageContexts{b: b}
+	if dockerfile != nil {
+		res, err := b.readAndParseDockerfile(dockerfile)
+		if err != nil {
+			return nil, err
+		}
+		b.dockerfile = res
+	}
```

Comment:
This was just removed in a recent PR. I think with the two suggestions above it wont be necessary anymore